src/extensions/__tests__/is-compatible-extension.test.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import semver from "semver";
+import {
+  isCompatibleExtension,
+} from "../extension-discovery/is-compatible-extension/is-compatible-extension";
+import type { LensExtensionManifest } from "../lens-extension";
+
+describe("Extension/App versions compatibility check", () => {
+  it("is compatible with exact version matching", () => {
+    expect(isCompatibleExtension({
+      appSemVer: semver.coerce("5.5.0"),
+    })(getExtensionManifestMock({
+      lensEngine: "5.5.0",
+    }))).toBeTruthy();
```

Comment:
Most tests duplicate this test logic. How about:

`expect(versionsAreCompatible("5.5.0", "5.6.0")).toBe(false)`;

Reply:
@jansav sounds good. I'll create follow-up with that test update.
But we compare not extension's version ( `versionOfExtension`), but but it's `engines.lens` version.